### PR TITLE
unixPB: Add Docker support for SLES 12

### PR DESCRIPTION
**Pull request by aswinkr77**
Extends docker support to SLES 12.
Fixes: https://github.com/adoptium/infrastructure/issues/3326

All the relevant packages mentioned in the PB are available as part of `Containers Module 12` repo. eg: https://scc.suse.com/packages/25502818

note: docker not supported in sles 12 aarch64
##### Checklist

- [x] commit message has one of the [standard prefixes](https://github.com/adoptium/infrastructure/blob/master/CONTRIBUTING.md#commit-messages)
- [ ] [faq.md](https://github.com/adoptium/infrastructure/blob/master/FAQ.md) updated if appropriate
- [ ] other documentation is changed or added (if applicable)
- [ ] playbook changes run through [VPC](https://ci.adoptium.net/view/Tooling/job/VagrantPlaybookCheck/) or [QPC](https://ci.adoptium.net/view/Tooling/job/QEMUPlaybookCheck/) (if you have access)
- [ ] VPC/QPC not applicable for this PR
- [ ] for inventory.yml changes, bastillion/nagios/jenkins updated accordingly

### CODE REVIEW

**Comment by aswinkr77** [comment]
@Haroon-Khel FYI.

**Comment by sxa** [comment]
Can I ask what the use case is for wanting to do updates on SLES12? It gives us quite a few problems elsewhere and I've been considering not maintaining any of the SLES12 infrastructure.
FYI @steelhead31

**Comment by aswinkr77** [comment]
@sxa we(IBM) still uses a couple of SLES 12 SP5 for our tests and builds which uses Docker. So thought of pushing the Docker support into Adoptium first.

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Docker/tasks/sles.yml [hunk 1]
```diff
@@ -22,7 +24,18 @@
...
+
+- name: Install docker in SLES 12
+  package:
+    update_cache: yes
+    name: ['containerd', 'runc', 'docker']
+    state: latest
+  when:
+    - ansible_distribution == "SLES"
+    - ansible_distribution_major_version == "12"
+    - ansible_architecture != "aarch64"  #no docker support for sles 12 aarch64
```
Comment (sxa) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Docker/tasks/sles.yml:41]: Is this exclusion really only for aarch64? Will it install on s390x/ppc64le? If not we should switch this to be "ansible_architecture=x64"

Comment (aswinkr77) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Docker/tasks/sles.yml:41]: Yep. Only for aarch64. There is docker support in s390x and ppc64le.
ppc64le: https://scc.suse.com/packages/25502809
s390x: https://scc.suse.com/packages/25502803

Tested on x86, ppc64le and s390x.

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Docker/tasks/sles.yml [hunk 2]
```diff
@@ -2,14 +2,16 @@
 - name: Add Docker Repo for SLES15
   command: zypper ar https://download.docker.com/linux/sles/docker-ce.repo
   when:
+    - ansible_distribution_major_version == "15"
     - ansible_architecture == "x86_64" or ansible_architecture == "s390x"
 
-- name: Add security repo for container-selinux
-  community.general.zypper_repository:
+- name: Add security repo for container-selinux for SLES 15
+  zypper_repository:
```
Comment (sxa) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Docker/tasks/sles.yml:9]: What's the reason for removing the full qualification here? Did you find it didn't work?

Comment (aswinkr77) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Docker/tasks/sles.yml:9]: Full qualification was not working. This is the error I got:
```
couldn't resolve module/action 'community.general.zypper_repository`
```

I searched within the entire Adoptium PB. The same module is used in the `Common` role without the full name.
https://github.com/adoptium/infrastructure/blob/0077e292c4691e2f477f361d6535382dda938363/ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/SLES.yml#L9-L11

And also in the documentation for Ansible 2.9, it's also using `zypper_repository`. And after changing it to just the module name, it worked fine.

And to add this, the security repo for container-selinux for SLES 15 is no longer valid. 
The link doesn't exist anymore: https://download.opensuse.org/repositories/security:SELinux/SLE_15_SP3/security:SELinux.repo

Comment (sxa) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Docker/tasks/sles.yml:9]: Thanks for the explanation on the qualification of `zypper_repository`

> the security repo for container-selinux for SLES 15 is no longer valid.

FYI @Haroon-Khel - any thoughts on this?

**Review by steelhead31 — APPROVED** [review]
This looks ok, I do have concerns about SLES12 support in general, as there are issues with the default repositories, I believe you need a valid support/subscription to use the SLES12 repositories. Though that doesn't necessarily impact this change.

**Review by sxa — APPROVED** [review]

**Pull request merged**